lms/djangoapps/instructor_task/tasks_helper.py

```diff
@@ -1599,3 +1600,47 @@ def invalidate_generated_certificates(course_id, enrolled_students, certificate_
         download_url='',
         grade='',
     )
+
+
+def upload_ora2_data(
+        _xmodule_instance_args, _entry_id, course_id, _task_input, action_name
+):
+    """
+    Collect ora2 responses and upload them to S3 as a CSV
+    """
+
+    start_date = datetime.now(UTC)
+    start_time = time()
+
+    num_attempted = 1
+    num_total = 1
+
+    task_progress = TaskProgress(action_name, num_total, start_time)
+    task_progress.attempted = num_attempted
+
+    curr_step = {'step': "Collecting responses"}
+
+    task_progress.update_task_state(extra_meta=curr_step)
+
+    try:
+        header, datarows = OraAggregateData.collect_ora2_data(course_id)
+        rows = [header] + [row for row in datarows]
+    # Update progress to failed regardless of error type
+    except Exception:  # pylint: disable=broad-except
```

Comment:
I'm in favor of leaving this `except` in place, in case it's needed to diagnose prod issues.

Will this log the stack trace of the error? Is that easy/worthwhile to add?